### ISSUE TITLE
No Style for tooltip (master) & Font size in the tooltip (v0.1.4)

### ISSUE THREAD

**Issue by ddotlic** [Microsoft/TypeScript-Sublime-Plugin#219]
Hi,

just realized that with the latest official released version (0.1.4) the tooltips (which I've had for quite some time because I used the HEAD version of the master from github repo) are finally colorized (this did not work - in the sense that popup was black text on white - before and still doesn't if I just clone the repo, weird).

However, the font size is a bit small, let me try to demonstrate:

![fontsize](https://cloud.githubusercontent.com/assets/1737586/8034393/88587f0a-0de7-11e5-9660-e5a13ccecc81.png)

Is there a way to configure the font size?

**Comment by ddotlic** [comment]
LOL, I just looked in the package directory, is it really that simple... the popup is - or rather its template is - `popup.html`??

If so, I can temporarily patch this myself, but you should probably replace the font size by the one actually used in the editor. If you're already doing this (sorry for being ignorant) by, say, reducing the editor's font a little, then the code is a bit too aggressive.

It might actually be the best to simply let us choose the font size via preferences, then replace in the template html.

**Comment by ddotlic** [comment]
After some experimentation, changing `popup.html` does nothing to the final output. If it is used, it's not obvious how (at the moment can't look into the source code of plugin to figure this out on my own).

**Comment by zhengbli** [comment]
I too believe that the font size of the tooltip should be customizable. We will add that option soon. Appreciate your experiment with the plugin :)

**Comment by ddotlic** [comment]
@zhengbli No problem, I didn't really do much except tinker a tiny bit.

I also removed the bold font for the tooltip, maybe a better option would be to let people entirely (to the extent that template can/should be customizable) change the template? Not sure how that would work though because the file itself is kept in the plugin's folder and obviously gets rewritten when next update comes.

Thanks again for this plugin!

**Comment by zhengbli** [comment]
@ddotlic Sorry I just realize that you said you don't have the tooltip if cloning from the repo directly. That is not expected. Do you see any exceptions in the console?

**Comment by ddotlic** [comment]
I actually have tooltips if I clone the repo, but it has been black text on white ever since I switched to the source version, including now (just pulled latest on this machine).

Here's how it looks

![nocolor](https://cloud.githubusercontent.com/assets/1737586/8045092/412ff82a-0e31-11e5-9655-b3b9ff4ca227.png)

If I delete the folder and install 0.1.4 through Package Manager I can get color tooltips which respect the theme just fine (which is when I've started playing with popup.html to adjust the font size).

I don't want to push this issue (black on white) because most people will use the Package Manager version anyway.

**Comment by ddotlic** [comment]
And no, I'm not seeing anything of note in the console.

The last 4 lines were:
```
Package Control: No updated packages
Packages/TypeScript/popup.html
insert_snippet
insert_snippet
```

**Comment by zhengbli** [comment]
Looks like the problem only happens for 64 bit Sublime Text on Windows. Is that what you have?

**Comment by zhengbli** [comment]
I think the problem is that the win 64 version Sublime cannot handle CR line endings in the tooltip html string. Should be fixed in PR #222

**Comment by ddotlic** [comment]
Indeed, I am running 64bit "portable" version on Windows 8.1. Will try the fix once merged in master.

Strangely, installing normally (on the same 64bit Windows) works fine.

**Comment by zhengbli** [comment]
It looks like the ``popup.html`` in the repo and the ``popup.html`` in the package control has different line endings. I guess it's due to git configuration, although from the history of that file nobody has changed it...

**Comment by ddotlic** [comment]
Ah, yes of course, the `autocrlf` and friends... If only we (the whole industry) could forgo endless little issues due to the miniscule platform differences, sigh.
I've noticed that the tooltip has slightly different - shall I say better? - coloring than the code; please take a look at my first screenshot and note that `AnnotationPoint` is not colored light blue like most other types in other methods in the editor itself. Is that intentional? Are you still working on/trying to better colorize the text in the editor (I know sublime's approach is mainly regex based so not great)? Did you see that in the recent versions of sublime there should be reportedly [a better way to describe syntax](http://www.sublimetext.com/docs/3/syntax.html) in `.sublime-syntax` files? Any chance on switching to this or using it in combination with `tmLanguage` (latest builds seem to add this option of combining the two)? Sorry for the barrage of questions :wink:

**Comment by zhengbli** [comment]
At the moment we are not trying to take over Sublime's own syntax rendering process (which might be possible but may affect performance greatly, as there is not really ways to define a code-based grammer like what Atom offers). The sublime's approach does have some issues, especially with line delimiters, but in most cases it is working fine. We will definitely looking into the new syntax format, however the comparability with Sublime 2 may be a concern. We'll explore and see I guess.

BTW, the fix is merged into master, if you can give it a shot. Thanks!

**Comment by ddotlic** [comment]
@zhengbli Works like a charm, great.

**Comment by ddotlic** [comment]
@hoanhtien Hey, I think the reason this was kept open was to track eventual improvement where the template for the tooltip (notably the font size) can be either customized by the end user or at least so that the template does not hardcode the font size but uses sublime's (user's) settings.

Was this done or is closing this issue just bookkeeping? Thanks!

**Comment by ddotlic** [comment]
I've renamed the item to better reflect the remaining work, thanks for reopening!
